cachi2/core/package_managers/cargo/main.py

```diff
@@ -8,4 +18,42 @@ def fetch_cargo_source(request: Request) -> RequestOutput:
     environment_variables: list[EnvironmentVariable] = []
     project_files: list[ProjectFile] = []
 
+    for package in request.cargo_packages:
+        package_dir = request.source_dir.join_within_root(package.path)
+        components.extend(_resolve_cargo_package(package_dir, request.output_dir))
+        # cargo allows to specify configuration per-package
+        # https://doc.rust-lang.org/cargo/reference/config.html#hierarchical-structure
+        project_files.append(_use_vendored_sources(package_dir))
+
     return RequestOutput.from_obj_list(components, environment_variables, project_files)
+
+
+def _resolve_cargo_package(package_dir: RootedPath, output_dir: RootedPath) -> list[Component]:
+    """Resolve a single cargo package."""
+    vendor_dir = output_dir.join_within_root("deps/cargo")
+    cmd = ["cargo", "vendor", "--locked", str(vendor_dir)]
+    log.info("Vendoring crates at %s", package_dir)
```

Comment:
nitpick: I forgot to comment on this log message during first round of reviews - please stick to something like "Fetching" or whatever wording we use across the other backends, the fact that we're vendoring is an internal implementation detail which hypothetically could change at some point in the future.